jobserver/authorization/roles.py

```diff
@@ -48,15 +48,14 @@ class StaffAreaAdministrator:
 
 
 class InteractiveReporter:
-    display_name = "Interactive Reporter"
-    description = """View analysis requests and reports for projects that used OpenSAFELY Interactive."""
-    models = [
-        "jobserver.models.project_membership.ProjectMembership",
-        "jobserver.models.user.User",
-    ]
-    permissions = [
-        release_file_view,
-    ]
+    """
+    Vestigial role class to support data migration to remove assignments of this role to users.
+    Not to be used outside of this migration.
+    Must be removed once this data migration is eliminated.
+    """
+
```

Comment:
You will need to keep the display_name here, otherwise you will get an error when viewing some users in the staff area (such as my Interactive only user).

The error is occurring because you've not removed the roles from ProjectMembership, but there might be something else I've missed.

If you decide to remove roles from ProjectMemberships in code, you'll need to use the project_members.remove(..) "command".
https://github.com/opensafely-core/job-server/blob/04a50694712f8b20e6130d2f3df3f7f425719df9/jobserver/commands/project_members.py#L74

Reply:
That's a good point, I'd forgotten about project level roles. 

However, this raises another question - who should the "by" be in this data migration? Me? Do we want/need the audit log to be correct for this operation?

Reply:
Actually, slight correction, I think the migration needs to call `update_roles` and not `remove`, because the latter also removes the project membership, which I don't think we want.

I think we want the audit log to be correct, but the question is what correct means in this case. We don't need to provide a valid username, but perhaps it's easier if it's you. We'll want to inform the IG team once the migration has been applied too.

Reply:
OK I've modified the migration accordingly (with a bit of _fun_ along the way, see the comment in the code). Could you take another look?

Reply:
Thanks. I've reapplied it locally and that seems to work fine. I've added a comment to clarify the comment. It does sound like it was a bit of a pain to write.